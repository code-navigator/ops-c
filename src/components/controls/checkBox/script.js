export default {
  name: 'checkBox',

  props: [
    'label',
    'value'
  ],

  computed: {
    input: {
      get () {
        return this.value
      },
      set (newValue) {
        this.$emit('keyup', newValue)
      }
    }
  }
}
