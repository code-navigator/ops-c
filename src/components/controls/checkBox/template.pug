//- Checkbox control
v-flex(xs12)
  v-checkbox(
    :label="label"
    v-model="input"
  )