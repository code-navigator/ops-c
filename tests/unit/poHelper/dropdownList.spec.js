import { createLocalVue, mount } from '@vue/test-utils'
import { TestHelpers } from './../testHelpers'
import checkBox from './../../../src/components/controls/dropdownlist'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuetify)
localVue.use(Vuex)

describe('test.vue', () => {
  let h
  let store
  let wrapper
  const label = 'some label'
  const items = ['item 1']
  const value = ''

  beforeEach(() => {
    store = new Vuex.Store({})
    wrapper = mount(checkBox, {
      localVue,
      store,
      propsData: {
        items,
        label,
        value
      }
    })
    h = new TestHelpers(wrapper, expect)
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('should have a name', () => {
    h.hasName('dropdownList')
  })

  it('should mount without errors', () => {
    h.isInstance()
  })

  it('should display a label', () => {
    h.hasText(label)
  })

  // it('should emit a "keyup" event when clicked', () => {
  //   const app = document.createElement('div')
  //   app.setAttribute('data-app', true)
  //   document.body.appendChild(app)
  //   h.noEmits('keyup')
  //   h.click('input')
  //   h.emits('keyup')
  // })

  it('should render correctly', () => {
    h.renders()
  })
})
