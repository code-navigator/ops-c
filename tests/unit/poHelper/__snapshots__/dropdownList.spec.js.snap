// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.vue should render correctly 1`] = `
<div class="flex xs12">
  <div class="v-input dropdownList v-text-field v-select v-autocomplete">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-menu v-menu--inline">
          <div class="v-menu__activator">
            <div class="v-select__slot">
              <label aria-hidden="true" class="v-label" style="left: 0px; position: absolute;">Select some label</label>
              <input aria-label="Select some label" type="text" autocomplete="off" role="combobox">
            </div>
          </div>
          <div class="v-menu__content v-autocomplete__content" style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;">
            <div class="v-select-list v-card">
              <div class="v-list v-list--dense">
                <div>
                  <a class="v-list__tile v-list__tile--link">
                    <div class="v-list__tile__content">
                      <div class="v-list__tile__title">item 1</div>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages">
          <div class="v-messages__wrapper"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
